app/scripts/services/dimStoreService.factory.js

```diff
@@ -1593,5 +1602,22 @@
       return ret;
     }
     // code above is from https://github.com/DestinyTrialsReport
+
+    function loadJSON(file) {
+      return $http.get(file)
+               .then(function(json) {
+                 return json.data.itemHash;
+               });
+    }
+
+    function getMissingSourceHashes(itemHash, missingSources) {
+      if (missingSources.$$state.value) {
```

Comment:
This isn't the right way to use a promise. You want to resolve it using `then` (or add it as an argument to a `$q.all`).